backend/infrahub/services/scheduler.py

```diff
@@ -54,32 +58,29 @@ async def initialize(self, service: InfrahubServices) -> None:
 
     async def start_schedule(self) -> None:
         for schedule in self.schedules:
-            asyncio.create_task(
-                run_schedule(schedule=schedule, service=self.service), name=f"scheduled_task_{schedule.name}"
-            )
+            asyncio.create_task(self.run_schedule(schedule=schedule), name=f"scheduled_task_{schedule.name}")
 
     async def shutdown(self) -> None:
         self.running = False
 
+    async def run_schedule(self, schedule: Schedule) -> None:
+        """Execute the task provided in the schedule as per the defined interval
 
-async def run_schedule(schedule: Schedule, service: InfrahubServices) -> None:
-    """Execute the task provided in the schedule as per the defined interval
-
-    Once the service is marked to be shutdown the scheduler will stop executing tasks.
-    """
-    for _ in range(schedule.start_delay):
-        if not service.scheduler.running:
-            return
-        await asyncio.sleep(delay=1)
-
-    service.log.info("Started recurring task", task=schedule.name)
-
-    while service.scheduler.running:
-        try:
-            await schedule.function(service)
-        except Exception as exc:  # pylint: disable=broad-exception-caught
-            service.log.error(str(exc))
-        for _ in range(schedule.interval):
-            if not service.scheduler.running:
+        Once the service is marked to be shutdown the scheduler will stop executing tasks.
+        """
+        for _ in range(schedule.start_delay):
+            if not self.running:
                 return
             await asyncio.sleep(delay=1)
+
+        assert self.service is not None, "InfrahubScheduler.service is None"
```

Comment:
We should avoid to use assert within the production code and instead raise errors.